op-node/rollup/interop/managed/system.go

```diff
@@ -341,6 +346,54 @@ func (m *ManagedMode) Reset(ctx context.Context, lUnsafe, xUnsafe, lSafe, xSafe,
 	return nil
 }
 
+// scanL2ForLatestLocalUnsafe scans the op-node's L2 chain starting from l2Unsafe (which we know is valid because it's given by the supervisor)
+// and check until the latestUnsafe block.
+func (m *ManagedMode) scanL2ForLatestLocalUnsafe(ctx context.Context, l2Unsafe eth.BlockID) (eth.L2BlockRef, error) {
+	valid, err := m.l2.L2BlockRefByNumber(ctx, l2Unsafe.Number)
```

Comment:
I think it is safer to fetch blocks using hash, not number for handling reorg error cases.

Reply:
I think in this particular case, it doesn't matter if we fetch by the `number` or `hash`, but I can change it.